src/model_loader_tinygltf.cpp

```diff
@@ -0,0 +1,291 @@
+			ModelMeshData* mesh_data = new ModelMeshData();
+
+			{ // GET INDICES
+				auto idx_accessor = tg_model.accessors[primitive.indices];
+
+				const auto& buffer_view = tg_model.bufferViews[idx_accessor.bufferView];
+				const auto& buffer = tg_model.buffers[buffer_view.buffer];
+				const auto data_address = buffer.data.data() + buffer_view.byteOffset + idx_accessor.byteOffset;
+				const auto byte_stride = idx_accessor.ByteStride(buffer_view);
+
+				mesh_data->m_indices.resize(idx_buffer_offset + idx_accessor.count);
+
+				memcpy(mesh_data->m_indices.data() + idx_buffer_offset, data_address, idx_accessor.count * byte_stride);
+				idx_buffer_offset += idx_accessor.count * byte_stride;
+			}
+
+			// Get other attributes
+			for (const auto& attrib : primitive.attributes)
+			{
+				const auto attrib_accessor = tg_model.accessors[attrib.second];
+				const auto& buffer_view = tg_model.bufferViews[attrib_accessor.bufferView];
+				const auto& buffer = tg_model.buffers[buffer_view.buffer];
+				const auto data_address = buffer.data.data() + buffer_view.byteOffset + attrib_accessor.byteOffset;
+				const auto byte_stride = attrib_accessor.ByteStride(buffer_view);
+				const auto count = attrib_accessor.count;
+				if (attrib.first == "POSITION") {
+					mesh_data->m_positions.resize(position_buffer_offset + count);
+					mesh_data->m_colors.resize(position_buffer_offset + count);
+
+					memcpy(mesh_data->m_positions.data() + position_buffer_offset, data_address, count * byte_stride);
+					position_buffer_offset += count * byte_stride;
+				}
+				else if (attrib.first == "NORMAL") {
+					mesh_data->m_normals.resize(normal_buffer_offset + count);
+
+					memcpy(mesh_data->m_normals.data() + normal_buffer_offset, data_address, count * byte_stride);
+					normal_buffer_offset += count * byte_stride;
+				}
+				else if (attrib.first == "TEXCOORD_0") {
+					std::vector<DirectX::XMFLOAT2> f2_data;
+					f2_data.resize(count);
+
+					memcpy(f2_data.data(), data_address, count * byte_stride);
+
+					for (auto& f2 : f2_data)
+					{
+						mesh_data->m_uvw.push_back({ f2.x, f2.y*-1, 0 });
+					}
+
+					uv_buffer_offset += count * sizeof(DirectX::XMFLOAT3);
+				}
+			}
+
+			auto tangent_bitangent = ComputeTangents(mesh_data);
+			mesh_data->m_tangents = tangent_bitangent.first;
+			mesh_data->m_bitangents = tangent_bitangent.second;
+			mesh_data->m_uvw.resize(mesh_data->m_positions.size());
+			mesh_data->m_material_id = primitive.material;
+
+			model->m_meshes.push_back(mesh_data);
+		}
+	}
+
+	ModelData* TinyGLTFModelLoader::LoadModel(std::string_view model_path)
+	{
+		tinygltf::Model tg_model;
+		tinygltf::TinyGLTF loader;
+		std::string err;
+		std::string warn;
+		std::string path = std::string(model_path);
+
+		if (!loader.LoadASCIIFromFile(&tg_model, &err, &warn, path))
+		{
+			LOGC("TinyGLTF Parsing Failed");
+		}
+		//bool ret = loader.LoadBinaryFromFile(&model, &err, &warn, argv[1]); // for binary glTF(.glb)
```

Comment:
Unused code, might as well move it unless it's really convenient to keep it around.